### PR TITLE
feat: set default trade assets based on connected EVM chain

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

- Sets the default trade assets based on the connected EVM chain in the portfolio view
- Sets the default sell asset to the asset's native fee asset

Screen Shot 2022-07-25 at 5 32 57 pm

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Closes https://github.com/shapeshift/web/issues/2222

## Risk

Touches the swapper and default trade assets. Ensure no regressions for trade functionality on the Ethereum EVM and other chains.

## Testing

- When connected to another EVM chain, say, Avalanche, the default assets should be set accordingly in the portfolio view.
- When on an asset route, that asset's native fee asset should be selected as the sell asset.

For Avalanche this is set to a `AVAX:wETH on Avalanche` trade pair.

## Screenshots (if applicable)

See description.

### CODE REVIEW

File: src/components/Trade/hooks/useTradeRoutes/useTradeRoutes.ts [hunk 1]
```diff
@@ -30,7 +33,23 @@ export const useTradeRoutes = (
     state: { wallet },
   } = useWallet()
 
-  const [defaultSellAssetId, defaultBuyAssetId] = getDefaultPair()
+  // fixme: abstract to custom hook
```
Comment (gomesalexandre) [inline src/components/Trade/hooks/useTradeRoutes/useTradeRoutes.ts:36]: Leaving this here as we want to abstract this away and are starting to duplicating the logic: We had some discussion about the naming of the ChainReference part, which in the context of ETH JSON-RPC ChainId is currently referred to as `chainId` / `evmChainId`.
Going forward, we will want to consolidate the naming, most likely as `ethNetwork`  or `ethNetworkVersion`. That will make this (as an abstracted hook then) a bit less confusing, not referring to `chainId` at all.

Comment (0xApotheosis) [reply src/components/Trade/hooks/useTradeRoutes/useTradeRoutes.ts]: Sounds good - used "eth network" terminology in https://github.com/shapeshift/web/pull/2223/commits/12582d02b2c0494078c59fb7fb37cdf2cf3a3459.

**Review by gomesalexandre — APPROVED** [review]
LGTM!

- In the dashboard view, the default pair is correctly set to ETH / FOX for Mainnet, and AVAX / WETH for Avalanche C-Chain. Tried to switch back and forth between the two a few times and this is reactive
- When on the asset page of a native asset (ETH/AVAX), the default buy asset is used (FOX  for ETH, WETH for AVAX)
- When on the asset page of some ERC20/.e, the sell asset is currently set to the native asset of the selected `ethNetwork` i.e ETH for Ethereum, AVAX for the Avalanche C-Chain.

The former is reactive on chain switch, the latter two are working consistently independently of the selected chain (tried switching chains back and forth to test for wrong reconciliation here, it works as it should!)

🐐

**Review by 0xdef1cafe — APPROVED** [review]
nice and tidy, does what it says on the box.

outside the scope of the PR - as we work more with these wallets (metamask & metamask imposters) that require a chain switch and force us to reset accountSpecifiers, we should look at making our logic more forgiving there such that we don't kick the user out to the connect wallet page on a chain switch. 

this needs discussion before implementation though, as it gets convoluted and tricky to detect/ignore certain events and prevent regressions.

**Pull request merged**